### PR TITLE
Frame transforms

### PR DESCRIPTION
**Pull request by aaronchongth**
* added some math functions to deal with transforms

* added dependency to `Eigen`

* transforms fleet robot coordinates into global `rmf_core` frames before publishing ROS 2 `FleetState`

* transforms global `rmf_core` request coordinates into local fleet robot frames when received over ROS 2 before commanding robots over DDS

* did some basic testing using the launch file `test_server.launch.py` with configuration `test_server.yaml`

### CODE REVIEW

File: servers/ros2/free_fleet_server/src/Server.cpp [hunk 1]
```diff
@@ -143,6 +144,10 @@ bool Server::is_ready()
 
 void Server::start()
 {
+  fleet_to_rmf_transform = math::convert(server_config.transformation);
+  fleet_to_rmf_yaw = fleet_to_rmf_transform.eulerAngles(0, 1, 2);
```
Comment (cnboonhan94) [inline servers/ros2/free_fleet_server/src/Server.cpp]: this causes a build error for me. replacing it with 
```
fleet_to_rmf_yaw = fleet_to_rmf_transform.eulerAngles(0, 1, 2)(0);
```
allows it to build. fleet_to_rmf_yaw is type `double`, but `.eulerAngles` returns a `Vector3f`

Comment (aaronchongth) [reply servers/ros2/free_fleet_server/src/Server.cpp]: gotcha, my bad

File: clients/ros1/free_fleet_client/src/Client.cpp [hunk 2]
```diff
@@ -437,6 +437,31 @@ void Client::read_requests()
...
+      ReadLock robot_transform_lock(robot_transform_mutex);
+      const double dx =
+          path_request->path._buffer[0].x
+          - current_robot_transform.transform.translation.x;
+      const double dy =
+          path_request->path._buffer[0].y
+          - current_robot_transform.transform.translation.y;
+      const double dist_to_first_waypoint = sqrt(dx*dx + dy*dy);
+      ROS_INFO("distance to first waypoint: %.2f\n", dist_to_first_waypoint);
+      const double max_dist_to_first_waypoint = 1.0;  // parameterize?
```
Comment (cnboonhan94) [inline clients/ros1/free_fleet_client/src/Client.cpp]: Perhaps this can be larger?

**Comment by cnboonhan94** [comment]
Briefly tested and seems to be correct implementation of transforms now. However it seems that the robot no longer moves through every Location in a PathRequest from HumanRMFCore, instead going straight to the final Location.

**Comment by aaronchongth** [comment]
> @aaronchongth can you add a ROS parameter in the client for `max_dist_to_first_waypoint` ? This is a helpful sanity-check when driving the client from upstream automation, but from human-guided-testing button panels, it's nice to have a lot more tolerance on the starting point. So it would be nice to be able to specify and/or override `max_dist_to_first_waypoint` on startup as a ROS1 param

gotcha, it should be up, waiting to be tested on the clients, creating another branch on `rmf_dp2` for the launch files

**Comment by aaronchongth** [comment]
> looks like the `test_server.yaml` is still conflicted with the `<<<< HEAD` and `>>>> master` stuff in it. The "new" transformation scheme is the `translation_x ... scale` , the "old" stuff is `transformation: [...]` and should be deleted.

yup, was too quick to commit, its fixed now.

**Pull request merged**